src/plugins/daoModule/index.ts

```diff
@@ -0,0 +1,214 @@
+import { getAddress, isAddress } from '@ethersproject/address';
+import { BigNumber, isBigNumberish } from '@ethersproject/bignumber/lib/bignumber';
+import { isHexString } from '@ethersproject/bytes';
+import { Contract } from '@ethersproject/contracts';
+import { JsonRpcProvider } from '@ethersproject/providers';
+import { keccak256 as solidityKeccak256 } from '@ethersproject/solidity';
+import { multicall, call, sendTransaction } from '../../utils';
+import getProvider from '../../utils/provider';
+import { _TypedDataEncoder } from "@ethersproject/hash";
+
+const EIP712_TYPES = {
+  "Transaction": [
+    {
+      "name": "to",
+      "type": "address"
+    },
+    {
+      "name": "value",
+      "type": "uint256"
+    },
+    {
+      "name": "data",
+      "type": "bytes"
+    },
+    {
+      "name": "operation",
+      "type": "uint8"
+    },
+    {
+      "name": "nonce",
+      "type": "uint256"
+    }
+  ]
+}
+
+const ModuleAbi = [
+  // Events
+  "event ProposalQuestionCreated(bytes32 indexed questionId, string indexed proposalId)",
+
+  // Read functions
+  "function executor() view returns (address)",
+  "function oracle() view returns (address)",
+  "function questionCooldown() view returns (uint32)",
+  "function buildQuestion(string proposalId, bytes32[] txHashes) view returns (string)",
+  "function executedPropsalTransactions(bytes32 questionHash, bytes32 txHash) view returns (bool)",
+
+  // Write functions
+  "function addProposal(string proposalId, bytes32[] txHashes)",
+  "function executeProposal(bytes32 questionId, string proposalId, bytes32[] txHashes, address to, uint256 value, bytes data, uint8 operation, uint256 nonce)",
+];
+
+const OracleAbi = [
+  // Read functions
+  "function resultFor(bytes32 question_id) view returns (bytes32)",
+  "function getFinalizeTS(bytes32 question_id) view returns (uint32)",
+];
+
+
+export interface ModuleTransaction {
+  to: string,
+  value: string,
+  data: string,
+  operation: string,
+  nonce: string
+}
+
+
+export interface ProposalDetails {
+  dao: string,
+  oracle: string,
+  cooldown: number,
+  proposalId: string,
+  questionId: string | undefined,
+  executionApproved: boolean,
+  finalizedAt: number | undefined,
+  nextTxIndex: number | undefined,
+  transactions: ModuleTransaction[],
+  txHashes: string[]
+}
+
+const buildQuestion = async (proposalId: string, txHashes: string[]) => {
+  const hashesHash = solidityKeccak256(["bytes32[]"], [txHashes]).slice(2);
+  return `${proposalId}␟${hashesHash}`
+}
+
+const getNextTxIndex = async (provider: JsonRpcProvider, moduleAddress: string, proposalId: string, txHashes: string[]) => {
+  // Call syntax is [to, [params]]
+  const question = await buildQuestion(proposalId, txHashes)
+  const questionHash = solidityKeccak256(["string"], [question])
+  const executionStati = (await multicall(provider.network.chainId.toString(), provider, ModuleAbi,
```

Comment:
Some network keys are not the same as their chainId, it would be better to pass the network string in the function param directly (same goes for others functions in the plugin).

```suggestion
  const executionStati = (await multicall(network, provider, ModuleAbi,
```